File: sources/java/04_array_sum/Main.java
// task 04 array_sum — expected output: 499999500000
// build: javac Main.java    run: java Main
// build (graalvm native-image): native-image -O2 Main    run: ./Main

public class Main {
    public static void main(String[] args) {
        final int n = 1000000;
        long[] array = new long[n];
        for (int i = 0; i < n; i++) {
            array[i] = i;
        }
        long total = 0;
        for (int i = 0; i < n; i++) {
            total += array[i];
        }
        System.out.println(total);
    }
}

File: sources/kotlin/04_array_sum/main.kt
// task 04 array_sum — expected output: 499999500000
// build: kotlinc main.kt -include-runtime -d prog.jar    run: java -jar prog.jar    [native build: kotlinc-native -opt -o prog main.kt    native run: ./prog]

fun main() {
    val n = 1000000
    val array = IntArray(n)
    for (i in 0 until n) {
        array[i] = i
    }
    var total = 0L
    for (i in 0 until n) {
        total += array[i]
    }
    println(total)
}
